pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -399,9 +408,35 @@ async def _map_user_prompt(self, part: UserPromptPart) -> list[PartDict]:
                 elif isinstance(item, BinaryContent):
                     # NOTE: The type from Google GenAI is incorrect, it should be `str`, not `bytes`.
                     base64_encoded = base64.b64encode(item.data).decode('utf-8')
-                    content.append({'inline_data': {'data': base64_encoded, 'mime_type': item.media_type}})  # type: ignore
+                    if item.vendor_metadata:
+                        video_metadata = VideoMetadata(
+                            fps=item.vendor_metadata.google_fps,
+                            start_offset=item.vendor_metadata.google_start_offset,
+                            end_offset=item.vendor_metadata.google_end_offset,
+                        )
+                        content.append(
+                            {
+                                'inline_data': {'data': base64_encoded, 'mime_type': item.media_type},
+                                'video_metadata': video_metadata,  # type: ignore
+                            }
+                        )
+                    else:
+                        content.append({'inline_data': {'data': base64_encoded, 'mime_type': item.media_type}})  # type: ignore
```

Comment:
Instead of repeating `'inline_data': {'data': base64_encoded, 'mime_type': item.media_type`, could we move this into a separate dict that we add `video_metadata` on to only if there is `item.vendor_metadata`?